client/checkout/blocks/index.js

```diff
@@ -157,15 +157,17 @@ if ( getUPEConfig( 'isWooPayEnabled' ) ) {
 	}
 }
 
-if ( getUPEConfig( 'isTokenizedCartPrbEnabled' ) ) {
-	registerExpressPaymentMethod(
-		tokenizedCartPaymentRequestPaymentMethod( api )
-	);
-} else if ( getUPEConfig( 'isExpressCheckoutElementEnabled' ) ) {
-	registerExpressPaymentMethod( expressCheckoutElementApplePay( api ) );
-	registerExpressPaymentMethod( expressCheckoutElementGooglePay( api ) );
-} else {
-	registerExpressPaymentMethod( paymentRequestPaymentMethod( api ) );
+if ( getUPEConfig( 'isPaymentRequestEnabled' ) ) {
+	if ( getUPEConfig( 'isTokenizedCartPrbEnabled' ) ) {
+		registerExpressPaymentMethod(
+			tokenizedCartPaymentRequestPaymentMethod( api )
+		);
+	} else if ( getUPEConfig( 'isExpressCheckoutElementEnabled' ) ) {
+		registerExpressPaymentMethod( expressCheckoutElementApplePay( api ) );
+		registerExpressPaymentMethod( expressCheckoutElementGooglePay( api ) );
+	} else {
+		registerExpressPaymentMethod( paymentRequestPaymentMethod( api ) );
+	}
 }
```

Comment:
This is a matter of preference, but I started to find the below approach more readable after some time in our back-end PHP:

```suggestion
if (!getUPEConfig('isPaymentRequestEnabled')) {
    return;
}

if (getUPEConfig('isTokenizedCartPrbEnabled')) {
    registerExpressPaymentMethod(tokenizedCartPaymentRequestPaymentMethod(api));
    return;
}

if (getUPEConfig('isExpressCheckoutElementEnabled')) {
    registerExpressPaymentMethod(expressCheckoutElementApplePay(api));
    registerExpressPaymentMethod(expressCheckoutElementGooglePay(api));
    return;
}

registerExpressPaymentMethod(paymentRequestPaymentMethod(api));
```

feel free to ignore it if the change does not seem convincing.

Reply:
It is indeed more readable, but we have more code after that we want to execute regardless of PRB/ECE settings. If we want to early return we need to create a new function for this so it doesn't affect the rest of the code. But then I think it starts to get less readable, what do you think?

```suggestion
const maybeRegisterExpressCheckoutMethod = () => {
	if (!getUPEConfig('isPaymentRequestEnabled')) {
	    return;
	}
	
	if (getUPEConfig('isTokenizedCartPrbEnabled')) {
	    registerExpressPaymentMethod(tokenizedCartPaymentRequestPaymentMethod(api));
	    return;
	}
	
	if (getUPEConfig('isExpressCheckoutElementEnabled')) {
	    registerExpressPaymentMethod(expressCheckoutElementApplePay(api));
	    registerExpressPaymentMethod(expressCheckoutElementGooglePay(api));
	    return;
	}
	
	registerExpressPaymentMethod(paymentRequestPaymentMethod(api));
}
maybeRegisterExpressCheckoutMethod();
```

Reply:
Yeah, agreed, I'd keep it as it is then without changing your final version 👍 🚢